jsonrpc/jsonrpc_test.go

```diff
@@ -0,0 +1,25 @@
+package jsonrpc
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSendJsonRpcRequest(t *testing.T) {
+	addr := resetMockRelay()
+	req := NewJSONRPCRequest(1, "eth_call", "0xabc")
+	res, err := req.Send(addr)
+	assert.Nil(t, err, err)
+
+	reply := new(string)
+	err = json.Unmarshal(res.Result, reply)
+	assert.Nil(t, err, err)
+	assert.Equal(t, "0x12345", *reply)
+
+	req2 := NewJSONRPCRequest(2, "unknown", "foo")
+	res2, err := req2.Send(addr)
+	assert.Nil(t, err, err)
```

Comment:
It's not required, that's what's printed in case it's not nil. So adding the error another time will then print the error in case it exists. If you don't add it, it will not print it. So it's good to always add it at the end.